packages/amplify-appsync-simulator/src/velocity/util/opensearch-utils.ts

```diff
@@ -0,0 +1,631 @@
+/**
+ * Utility class to convert the given params to opensearch query DSL
+ */
+class ElasticsearchUtils {
+    private static readonly ONE: number = 1;
+    private static readonly BOOL: string = "bool";
+    private static readonly MUST: string = "must";
+    private static readonly MUST_NOT: string = "must_not";
+    private static readonly SHOULD: string = "should";
+    private static readonly MATCH: string = "match";
+    private static readonly MATCH_PHRASE: string = "match_phrase";
+    private static readonly MATCH_PHRASE_PREFIX: string = "match_phrase_prefix";
+    private static readonly MULTI_MATCH: string = "multi_match";
+    private static readonly EXISTS: string = "exists";
+    private static readonly WILDCARD: string = "wildcard";
+    private static readonly REGEXP: string = "regexp";
+    private static readonly RANGE: string = "range";
+    private static readonly GT: string = "gt";
+    private static readonly GTE: string = "gte";
+    private static readonly LT: string = "lt";
+    private static readonly LTE: string = "lte";
+    private static readonly MINIMUM_SHOULD_MATCH: string = "minimum_should_match";
+    private static readonly FIELD: string = "field";
+
+    /**
+     * Convert a field and a value into Elasticsearch "match" expression.
+     *
+     * @param fieldName
+     *      The operand that is indexed in Elasticsearch.
+     * @param value
+     *      The value in the expression.
+     * @return
+     *      The converted Elasticsearch expression. Returns null if fieldName is invalid.
+     */
+    public toEqExpression(fieldName: string, value: any): any {
+        if (!fieldName) {
+            return null;
+        }
+
+        const updatedFieldName: string = ((typeof value) === "string") ? (fieldName + ".keyword") : fieldName;
```

Comment:
What if customer is passing a string for integer field like "10"?

Reply:
It only depends on the type of the field and not on value passed. All `String` type fields are mapped to have `.keyword` postfixes in Elasticsearch to make some operations like "word searching" optimal.